src/item.cpp

```diff
@@ -4139,6 +4139,21 @@ const item &item::get_contained() const
     return contents.front();
 }
 
+bool item::handle_spillable_contents( player &p )
```

Comment:
Could be made a free function `bool handle_spillable_contents( player &p, item &it)` since item functions `is_bucket_nonempty` and `spill_contents` are publicly accessible. This does not need to be a member function.